fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -154,9 +151,9 @@ def _call_oracle(self, lambda_vec):
         if isinstance(self.constraints, ClassificationMoment):
             redY = 1 * (signed_weights > 0)
         else:
-            redY = self.y
+            redY = self.constraints.y_reformat
         redW = signed_weights.abs()
-        redW = self.n * redW / redW.sum()
+        redW = self.constraints.total_samples * redW / redW.sum()
```

Comment:
I have to say this usage of a different `y_reformat` and different `n` confuses me. Can we at least document what this is supposed to do?

Reply:
It's doing exactly what it was before, except that knowledge of the data has been pushed into the Moment, per @MiroDudik 's suggestion.

Reply:
This is the right behavior for now. In a future PR we may want to push the construction of `redY` and `redW` into `signed_weights`. But please keep as is for now.

Reply:
This is perhaps an area we should take some time on in future. I think that part of the back-and-forth we've been having with this PR are due to us not being quite sure what should be where.